share/smack/top.py

```diff
@@ -899,85 +978,145 @@ def verify_bpl(args):
-            command += ["/bopt:proverOpt:SOLVER=Yices2"]
 
     elif args.verifier == 'symbooglix':
-        command = ['symbooglix']
-        command += [args.bpl_file]
-        command += ["--file-logging=0"]
-        command += ["--entry-points=%s" % ",".join(args.entry_points)]
-        command += ["--timeout=%d" % args.time_limit]
-        command += ["--max-loop-depth=%d" % args.unroll]
+        command = invoke_symbooglix(args)
 
     if args.verifier_options:
         command += args.verifier_options.split()
 
     verifier_output = try_command(command, timeout=args.time_limit)
-    verifier_output = transform_out(args, verifier_output)
-    result = verification_result(verifier_output, args.verifier)
+    return process_verifier_output(args, verifier_output)
 
-    if args.json_file:
-        with open(args.json_file, 'w') as f:
-            f.write(json_output_str(result, verifier_output, args.verifier))
 
-    if result in VResult.ERROR:
-        error = error_trace(verifier_output, args.verifier)
+def thread_verify_bpl(args, args_to_add):
 
-        if args.error_file:
-            with open(args.error_file, 'w') as f:
-                f.write(error)
+    if 'verifier' in args_to_add:
+        if args_to_add['verifier'] is 'portfolio':
+            raise RuntimeError("portfolio is not a valid verifier"
+                               " specification within the portfolio configuration")
+        else:
+            print("Warning: SMACK is using argument verifier from"
+                  " the chosen portfolio configuration")
+            args.verifier = args_to_add['verifier']
+    else:
+        raise RuntimeError("verifier is a required argument"
+                           " in the portfolio configuration file")
 
-        if not args.quiet:
-            print(error)
+    if 'modular' in args_to_add:
+        if args.modular:
+            raise RuntimeError("argument modular specified in both command"
+                               " line and portfolio configuration")
+        else:
+            print("Warning: SMACK is using argument modular from"
+                  " the chosen portfolio configuration")
+            args.modular = args_to_add['modular']
+
+    if (args.verifier is not 'boogie') and args.modular:
+        raise RuntimeError("Incompatible arguments modular"
+                           " and non-boogie verifier were specified")
+
+    if 'verifier-options' in args_to_add:
+        if args.verifier_options is not '':
+            raise RuntimeError("argument verifier-options specified in both"
+                               " command line and portfolio configuration")
+        else:
+            print("Warning: SMACK is using argument verifier-options from"
+                  " the chosen portfolio configuration")
+            args.verifier_options = args_to_add['verifier-options']
+
+    if 'solver' in args_to_add:
+        if args.solver is not 'z3': # is there a better check than not default?
```

Comment:
This should be `if args.solver != 'z3':`

Reply:
Also, there should be two space before your comment. So '#' should be preceded by two spaces.